### PR TITLE
Beds 195/upgrade primevue

### PR DESCRIPTION
**Pull request by MauserBitfly**
This PR
- updates the primevue version to `4.0.4`
  - Replace `Dropdown` with `Select`
  - Create new `BcTabList` component for the `Tabs`component and replacing all usages of the old `TabList`
  - Replace `InputSwitch`with `ToggleSwitch`
  - Adapted to  a lot of PrimeVue class names changes
  - Adjusted default styles (as they now have more default values) of several comonents
- Fix small translation but in `NotificationsNetworkTable`

This PR did not:
- make use of CSS variables introduced with the new version, but rather overwrites them to get to status quo

### CODE REVIEW

File: frontend/assets/css/prime.scss [hunk 1]
```diff
@@ -25,12 +26,11 @@
 
   transition: background-color 0.2s, color 0.2s, border-color 0.2s;
 
-  &:not([data-secondary]) {
+  &:not([ddata-p-severity="secondary"]) {
```
Comment (marcel-bitfly) [inline frontend/assets/css/prime.scss:29]: I think there is a typo

Comment (MauserBitfly) [reply frontend/assets/css/prime.scss]: good catch, fixed ;-)

File: frontend/components/bc/toggle/MultiBarButton.vue [hunk 2]
```diff
@@ -48,22 +39,12 @@ const icon = computed(() => {
       </div>
     </template>
     <ToggleButton
-      v-model="selected"
-      class="bc-toggle"
-      :class="readonlyClass"
-      on-label="''"
-      off-icon="''"
+      v-model="selected" class="bc-toggle" :class="readonlyClass"
```
Comment (marcel-bitfly) [inline frontend/components/bc/toggle/MultiBarButton.vue:42]: nitpick: why is this all in one line now?

File: frontend/package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "lodash-es": "^4.17.21",
     "luxon": "^3.4.4",
     "pinia": "^2.1.7",
-    "primevue": "3.46.0",
+    "primevue": "^4.0.4",
```
Comment (marcel-bitfly) [inline frontend/package.json:18]: should we pin this version?

File: frontend/types/hashTabs.ts [hunk 4]
```diff
@@ -1,5 +1,12 @@
+import type { IconDefinition } from '@fortawesome/fontawesome-svg-core'
+import type { Component } from 'vue'
+
 export type HashTab = {
+  component?: Component,
   disabled?: boolean,
-  index: number,
+  icon?: IconDefinition,
+  index: string,
```
Comment (marcel-bitfly) [inline frontend/types/hashTabs.ts:8]: this should be a number

Comment (marcel-bitfly) [inline frontend/types/hashTabs.ts:8]:
```suggestion
  index: number,
```
this should stay a number

File: frontend/types/hashTabs.ts [hunk 5]
```diff
@@ -1,5 +1,12 @@
...
+  placeholder?: string,
+  title?: string,
 }
 export type HashTabs = Record<string, HashTab>
```
Comment (marcel-bitfly) [inline frontend/types/hashTabs.ts:12]: couldn't that be an array instead of an object, then we would spare us the struggle of sorting this

Comment (MauserBitfly) [reply frontend/types/hashTabs.ts]: originally it was easier with a Record to map the url hash ... but now that the PrimeVue tabs no longer use an index but a string as key it's much easier and cleaner with an array.

File: frontend/types/hashTabs.ts (hunk 4 above)
Comment (MauserBitfly) [reply frontend/types/hashTabs.ts]: renamed to key:string and refactored the whole thing once more

File: frontend/components/bc/tab/BcTabList.vue [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+const sortedTabs = computed(() => orderBy(Object.entries(props.tabs).map(([
+  key,
+  tab,
+]) => {
+  return {
+    key,
+    ...tab,
+  }
+}), tab => parseInt(tab.index)))
```
[the remark this answers is not in the document]
Comment (MauserBitfly) [reply frontend/components/bc/tab/BcTabList.vue]: sorting no longer needed

**Review by marcel-bitfly — APPROVED** [review]
nice work...thanks for taking over this hard task 💪

**Pull request merged**